cdap-ui/app/cdap/components/FieldLevelLineage/v2/TimeRangePicker/index.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, { useContext } from 'react';
+import withStyles, { StyleRules } from '@material-ui/core/styles/withStyles';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import T from 'i18n-react';
+import { TIME_OPTIONS } from 'components/FieldLevelLineage/store/Store';
+import ExpandableTimeRange from 'components/TimeRangePicker/ExpandableTimeRange';
+import { IContextState, FllContext } from 'components/FieldLevelLineage/v2/Context/FllContext';
+
+const PREFIX = 'features.FieldLevelLineage.v2.TimeRangePicker';
+
+// These styles came from components/TimeRangePicker/TimeRangePicker.scss
+const styles = (): StyleRules => {
+  return {
+    view: {
+      padding: 10,
+    },
+    timeRangeContainer: {
+      display: 'inline-block',
+      position: 'relative',
+      marginLeft: 10,
+      width: 400,
+    },
+    timePickerContainer: {
+      paddingLeft: 50,
+    },
+    // To do: Style menu
+    menu: {
+      getContentAnchorEl: 'null',
+      anchorOrigin: {
+        vertical: 'bottom',
+        horizontal: 'center',
+      },
+    },
+  };
+};
+
+function TimeRangePicker({ classes }) {
+  const { start, end, selection, setTimeRange, setCustomTimeRange } = useContext<IContextState>(
+    FllContext
+  );
+
+  const onSelect = (e: React.ChangeEvent<{ value: string }>) => {
+    const range = e.target.value;
+
+    if (range === TIME_OPTIONS[0]) {
+      renderCustomTimeRange();
```

Comment:
I don' think this is necessary. The `renderCustomTimeRange` already has a check whether to render or not based on the selection.